agentcfg/model_test.go

```diff
@@ -44,3 +45,26 @@ func TestNewDoc(t *testing.T) {
 		assert.Equal(t, Result{Source{Etag: "123", Settings: Settings{"sample_rate": "0.5"}}}, d)
 	})
 }
+
+func TestQueryMarshaling(t *testing.T) {
+	for _, tc := range []struct {
+		name  string
+		input string
+		out   string
+	}{
+		{name: "third_party",
+			input: `{"service":{"name":"auth-service","environment":"production"},"applied_by_agent":true}`,
+			out:   `{"service":{"name":"auth-service","environment":"production"},"applied_by_agent":true,"etag":""}`},
+		{name: "elastic_apm",
+			input: `{"service":{"name":"auth-service","environment":"production"},"etag":"1234"}`,
+			out:   `{"service":{"name":"auth-service","environment":"production"},"etag":"1234","applied_by_agent":null}`},
```

Comment:
`"applied_by_agent":null` shouldn't be sent, right? That'll go away once omitempty is inside the double quotes.